src/app/app.component.ts

```diff
@@ -13,9 +13,17 @@ import {filter} from 'rxjs/operators';
   styleUrls: ['app.component.scss']
 })
 export class AppComponent implements OnInit, AfterViewInit {
+  private static serviceWorkerResolve: Function;
+
   @ViewChild('appDrawer') appDrawer: ElementRef;
   private firstUseOfCurrentRoute: boolean = true;
 
+  static getServiceWorkerInitFactory() {
+    return new Promise<any>((resolve, reject) => {
+      AppComponent.serviceWorkerResolve = resolve;
```

Comment:
This will resolve immediately and trigger service worker registration immediately right? It would be nice for the example here to do something like wait 3-5 seconds and then register the service worker.

Reply:
Sure, I will update the sample.